### PR TITLE
Adding tests for complete coverage of Polynomial

### PR DESCRIPTION
**Pull request by CodeMaxx**
I'll keep amending the commit after adding more tests
Please comment if you think more tests are required in some place.

@isuruf @Sumith1896

### CODE REVIEW

**Comment by CodeMaxx** [comment]
@isuruf

**Comment by isuruf** [comment]
Tests are failing

**Comment by CodeMaxx** [comment]
@isuruf Thats a problem with Travis.... the tests are passing otherwise.

**Comment by isuruf** [comment]
I restarted Travis. Appveyor failures seems legit. Can you look into it?

**Comment by CodeMaxx** [comment]
@isuruf I'm not able to understand the error.

**Comment by isuruf** [comment]
See here, https://ci.appveyor.com/project/certik/symengine/build/1418/job/cw70oi8hp6hjmru2#L1760
It's complaining that `<<` is not defined for `RCP<const Symbol>` which is true

File: symengine/tests/basic/test_polynomial.cpp [hunk 1]
```diff
@@ -335,6 +468,28 @@ TEST_CASE("UnivariatePolynomial get_args", "[UnivariatePolynomial]")
...
+{
+    RCP<const Symbol> x = symbol("x");
+    RCP<const UnivariatePolynomial> a
+        = univariate_polynomial(x, {{0, 1}, {1, 2}, {2, 4}});
+    RCP<const UnivariatePolynomial> b
+        = univariate_polynomial(x, {{0, 2}, {1, 2}, {2, symbol("b")}});
+
+    REQUIRE(a->max_coef() == 4);
+    REQUIRE(not(a->max_coef() == 2));
+    REQUIRE(b->max_coef() == symbol("b"));
```
Comment (isuruf) [inline symengine/tests/basic/test_polynomial.cpp]: Use `eq` here.

Error is due to `Catch` uses macros to take the LHS and RHS separately and will try to print each side with `<<` to show that RHS and LHS are different if there are different.

Comment (CodeMaxx) [inline symengine/tests/basic/test_polynomial.cpp]: Actually `max_coeff()` returns `Expression` so I'll change `symbol` to `Expression` and use `==`

**Comment by CodeMaxx** [comment]
Working now.... https://ci.appveyor.com/project/CodeMaxx/symengine

**Comment by CodeMaxx** [comment]
@isuruf Ping

**Comment by isuruf** [comment]
Coverage was improved by 0.78%. Thanks.

**Pull request merged**